pkg/controller/supportbundlecollection/controller.go

```diff
@@ -277,6 +314,12 @@ func (c *Controller) syncSupportBundleCollection(key string) error {
 		}
 		return nil
 	}
+	// Update internal SupportBundleCollection status. This is triggered when Agent reports status.
+	// The event that SupportBundleCollection CR status update will not trigger this logic.
```

Comment:
I think the comments are not accurate and maybe unnecessary.
It could be also triggered by the scheduling for expiry handling, and it seems now the expiry handling would never be executed as it's in createInternalSupportBundleCollection. I think the function name `createInternalSupportBundleCollection` is not accurate and some logic should be moved to `updateStatus`.

Reply:
Remove the comments, and move the call of `updateStatus` out of `createInternalSupportBundleCollection`, and call `updateStatus` only in `syncSupportBundleCollection`